library/core/src/option.rs

```diff
@@ -835,8 +868,8 @@ impl<T> Option<T> {
             *self = Some(f());
         }
 
-        match *self {
-            Some(ref mut v) => v,
+        match self {
+            Some(v) => v,
```

Comment:
If we're changing this method anyway, might as well call `insert`:
```rust
match self {
    Some(v) => v,
    None => self.insert(f()),
}
```
(GitHub didn't let me add this as a change suggestion)